master/buildbot/www/hooks/github.py

```diff
@@ -242,6 +245,30 @@ def _get_commit_msg(self, repo, sha):
         msg = data.get('commit', {'message': 'No message field'})['message']
         return msg
 
+    @defer.inlineCallbacks
+    def _get_pr_files(self, repo, number):
+        """
+        Get Files that belong to the Pull Request
+        :param repo: the repo full name, ``{owner}/{project}``.
+            e.g. ``buildbot/buildbot``
+        :param number: the pull request number.
+        """
+        headers = {"User-Agent": "Buildbot"}
+        if self._token:
+            headers["Authorization"] = "token " + self._token
+
+        url = "/repos/{}/pulls/{}/files".format(repo, number)
+        http = yield httpclientservice.HTTPClientService.getService(
+            self.master,
+            self.github_api_endpoint,
+            headers=headers,
+            debug=self.debug,
+            verify=self.verify,
+        )
+        res = yield http.get(url)
```

Comment:
what happen in case of error (rate limiting for example)

Reply:
The same thing that would happen with the `_get_commit_msg`.